tidy3d/components/medium.py

```diff
@@ -143,11 +143,52 @@ class NonlinearSusceptibility(NonlinearSpec):
     """
 
     chi3: float = pd.Field(
-        ..., title="Chi3", description="Chi3 nonlinear susceptibility.", units="um^2 / V^2"
+        0, title="Chi3", description="Chi3 nonlinear susceptibility.", units="um^2 / V^2"
     )
 
 
-NonlinearSpecType = Union[NonlinearSusceptibility]
+class KerrAndTPA(NonlinearSpec):
+    """Specification adding an instantaneous nonlinear susceptibility to a medium,
+    as well as a non-instantaneous nonlinear susceptibility representing two-photon
+    absorption (TPA).
+
+    Note
+    ----
+    The nonlinear constitutive relation is solved iteratively; it may not converge
+    for strong nonlinearities. Increasing `numiters` can help with convergence.
+
+    Note
+    ----
+    For complex fields (e.g. when using Bloch boundary conditions), the nonlinearity
+    is applied separately to the real and imaginary parts, so that the above equation
+    holds when both E and :math:`P_{NL}` are replaced by their real or imaginary parts.
+    The nonlinearity is only applied to the real-valued fields since they are the
+    physical fields.
+
+    Note
+    ----
+    Different field components do not interact nonlinearly. For example,
+    when calculating :math:`P_{NL, x}`, we approximate :math:`|E|^2 \\approx |E_x|^2`.
+    This approximation is valid when the E field is predominantly polarized along one
+    of the x, y, or z axes.
+
+    Example
+    -------
+    >>> medium = Medium(permittivity=2, nonlinear_spec=KerrAndTPA(chi3=1, sigma3=1))
+    """
+
+    chi3: float = pd.Field(
+        0, title="Chi3", description="Chi3 nonlinear susceptibility.", units="um^2 / V^2"
+    )
+
+    sigma3: float = pd.Field(
+        0,
```

Comment:
are these able to be negative? if not, can make them `pd.NonNegativeFloat`

Reply:
chi3 can be negative. Negative sigma3 makes it active so yeah it would be good to at least warn in this case.

Reply:
Since we have `allow_gain` field in Medium, shall we only allow negative sigma3 when `allow_gain=True`? This is useful in backend to catch divergence.

Reply:
Yeah, that's a good suggestion. I'll add a validator to `AbstractMedium`, since that would have access to `allow_gain` and `nonlinear_spec`